### PR TITLE
Move Driver to Gen2 API of DepthAI

### PR DESCRIPTION
**Pull request by kunaltyagi**
Closes #43

### CODE REVIEW

**Comment by kunaltyagi** [comment]
@ardiya Please check 282fdae

Also should this PR close #43?

**Review by ardiya — APPROVED** [review]
Yes it will close the timestamp PR.
Shall we create a git tag for the depthai 1 version before merging?

**Comment by kota-rr** [comment]
Will these also be merged?

https://github.com/rapyuta-robotics/depthai_ros/pull/49
https://github.com/rapyuta-robotics/depthai_ros/pull/51

File: depthai_ros_driver/test/depthai_api.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        inp->out.link(cam->inputConfig);
+        cam->preview.link(out->input);
+    }
+};
+
+TEST_F(SimplePipeline, FilterByName) {
+    std::vector<NodeConstPtr> nodes = filterNodesByName(p, "ColorCamera");
+    EXPECT_EQ(nodes.size(), 1);
+
+    nodes = filterNodesByName(p, "ColorCamer");
```
Comment (kota-rr) [inline depthai_ros_driver/test/depthai_api.cpp]: `ColorCamera`

**Comment by kunaltyagi** [comment]
No. They will be merged after this PR.

File: depthai_ros_driver/src/conversion.cpp [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        dst[i * 3 + 0] = b;
+    }
+    for (int i = 0; i < WH; ++i) {
+        uint8_t g = src[i + WH * 1];
+        dst[i * 3 + 1] = g;
+    }
+    for (int i = 0; i < WH; ++i) {
+        uint8_t r = src[i + WH * 2];
+        dst[i * 3 + 2] = r;
+    }
```
Comment (kota-rr) [inline depthai_ros_driver/src/conversion.cpp:32]: I guess single loop like `interleaved2planar` is faster, but I'm not certain,

File: depthai_ros_driver/test/depthai_api.cpp (hunk 1 above)
Comment (kota-rr) [inline depthai_ros_driver/test/depthai_api.cpp:37]: `ColorCamera`

**Review by kota-rr — APPROVED** [review]

File: depthai_ros_driver/test/depthai_api.cpp (hunk 1 above)
Comment (kunaltyagi) [reply depthai_ros_driver/test/depthai_api.cpp]:
```suggestion
    nodes = filterNodesByName(p, "ColorCamera");
```

File: depthai_ros_driver/src/conversion.cpp (hunk 2 above)
Comment (kunaltyagi) [reply depthai_ros_driver/src/conversion.cpp:32]: For modern compilers, I think this would be a piece of cake

**Pull request merged**